### PR TITLE
HOTFIX: Fix reset integration test hangs on busy wait

### PR DESCRIPTION
**Pull request by guozhangwang**
1. Found two root causes of the failure: a) appID is not set at constructor time and hence causing the condition method to fail, 2) class static properties are only populated once and hence when SSL tests are executed before non-SSL tests, the later will fail.

2. A bunch of log4j improvements along the debugging experience.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by guozhangwang** [comment]
retest this please

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-test-coverage/205/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-test-coverage/220/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-test-coverage/258/

**Comment by guozhangwang** [comment]
retest this please

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-test-coverage/304/

**Comment by guozhangwang** [comment]
@mjsax @dguy ready for reviews.

File: streams/src/test/java/org/apache/kafka/streams/integration/AbstractResetIntegrationTest.java [hunk 1]
```diff
@@ -173,8 +177,8 @@ void prepareTest() throws Exception {
         prepareEnvironment();
 
         // busy wait until cluster (ie, ConsumerGroupCoordinator) is available
-        TestUtils.waitForCondition(consumerGroupInactiveCondition, TIMEOUT_MULTIPLIER * CLEANUP_CONSUMER_TIMEOUT,
-                "Test consumer group active even after waiting " + (TIMEOUT_MULTIPLIER * CLEANUP_CONSUMER_TIMEOUT) + " ms.");
+        TestUtils.waitForCondition(new ConsumerGroupInactiveCondition(), TIMEOUT_MULTIPLIER * CLEANUP_CONSUMER_TIMEOUT,
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/integration/AbstractResetIntegrationTest.java:184]: Why do we need to generate a new object each time?

File: streams/src/test/java/org/apache/kafka/streams/integration/ResetIntegrationTest.java [hunk 2]
```diff
@@ -43,6 +41,8 @@
     private static final String TEST_ID = "reset-integration-test";
 
     static {
+        sslConfig = null;
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/integration/ResetIntegrationTest.java]: Why this change instead of setting `null` initially?

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-test-coverage/352/

File: streams/src/test/java/org/apache/kafka/streams/integration/ResetIntegrationTest.java (hunk 2 above)
Comment (guozhangwang) [reply streams/src/test/java/org/apache/kafka/streams/integration/ResetIntegrationTest.java]: The `sslConfig` is a static object of the underlying `AbstractResetIntegrationTest`, so in jenkins if `ResetIntegrationWithSSLTest` is triggered first `sslConfig` will be kept not null. This is from my observation in jenkins builds.

File: streams/src/test/java/org/apache/kafka/streams/integration/AbstractResetIntegrationTest.java (hunk 1 above)
Comment (guozhangwang) [reply streams/src/test/java/org/apache/kafka/streams/integration/AbstractResetIntegrationTest.java:184]: We cannot create an object before any tests are run, because `appID` is not initialized and that would cause test to pause. This is actually the root cause of the failure before.

**Review by mjsax — APPROVED** [review]
LGTM.

**Comment by mjsax** [comment]
Retest integration test fails in all builds... Seems to be related.

**Comment by guozhangwang** [comment]
> Retest integration test fails in all builds... Seems to be related.

Local runs with exactly the same parallelism / settings cannot reproduce the timeout exception, will trigger another round to double check.

**Comment by mjsax** [comment]
Happened again for one run. Maybe we need to bump up some timeout? Seems to be too unstable to get merged atm. WDYT?

**Comment by guozhangwang** [comment]
retest this please

**Comment by mjsax** [comment]
retest this please

**Comment by guozhangwang** [comment]
retest this please

**Comment by guozhangwang** [comment]
retest this please

**Comment by guozhangwang** [comment]
retest this please

**Comment by guozhangwang** [comment]
@mjsax The test passed three consecutive times on jenkins, which is 9 runs on different versions. Previously it fails almost on every run.

I will update the description and then merge it as is after removing all the debugging changes.

File: streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java [hunk 3]
```diff
@@ -104,7 +104,7 @@ public void start() throws IOException, InterruptedException {
 
         for (int i = 0; i < brokers.length; i++) {
             brokerConfig.put(KafkaConfig$.MODULE$.BrokerIdProp(), i);
-            log.debug("Starting a Kafka instance on port {} ...", brokerConfig.getProperty(KafkaConfig$.MODULE$.PortProp()));
+            log.debug("Starting a Kafka instance on port {} ...", brokerConfig.get(KafkaConfig$.MODULE$.PortProp()));
```
Comment (guozhangwang) [inline streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java:107]: This change is intentional: `getProperty` will always return null for non-string typed value, changed to `get` would return the correct int value for printing.
